ebu/benchmarkstt
# Conferatur (basic structure, docs, etc)

[![Build Status](https://travis-ci.com/MikeSmithEU/ai-benchmarking-stt.svg?branch=conferatur)](https://travis-ci.com/MikeSmithEU/ai-benchmarking-stt)

Things done in this version:
- setup basic structure
- created some normalisers
- created sphinx docs (see https://conferatur.mikesmith.eu/ )
- created a command line tool
- created proper pip setup for easy installation
- created manpage for command line tool
- created jsonrpc api
- make jsonrpc available through gunicorn (+ docker if wanted)
- added bash completion for command line tool
- added testing via pytest
- added Dockerfile + documentation
- moved/merged md docs into sphinx docs where applicable
- wrote csv and config file parser that supports comments (fixes the "known bugs" mentioned below)
- add tox and travis-ci conf 
- added a jsonrpc explorer 
   - added pre-filled examples
- added examples to each normaliser
- added normalizer logs
   - added possibility to get logs from api call
   - show logs from api call in jsonrpc explorer

todos:
- !!!! fix bug: only returns logs in api call when server running in log_level info
- could use some more cleanup
- maybe add comments to the more complex code, will probably be part of this PR as questions come in...
- create more unittests
- jsonrpc explorer: 
  - make encoding multiselect, or remove from api since this only applies to local (i.e. server) files (?)
- ...

~~known bugs, will be fixed with writing a custom own csv reader (didn't do that for now as this is more to show rudimentary structure):~~
~~- "#test" is incorrectly seen as a comment, if it is quoted, it should not be seen as a comment~~
~~- for config: if ending a line with a space, it will add a '' as last argument, need to pre-trim line before splitting into arguments~~

(replaces PR #16 avoiding git history rewrite and showing proper diff)
